packages/core/content-releases/server/src/bootstrap.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import type { LoadedStrapi } from '@strapi/types';
+import type { Entity as StrapiEntity } from '@strapi/types';
+import { RELEASE_ACTION_MODEL_UID } from './constants';
+
+const { features } = require('@strapi/strapi/dist/utils/ee');
+
+export const bootstrap = async ({ strapi }: { strapi: LoadedStrapi }) => {
+  if (
+    features.isEnabled('cms-content-releases') &&
+    strapi.features.future.isEnabled('contentReleases')
+  ) {
+    strapi.db.lifecycles.subscribe({
+      afterDelete(event) {
+        // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
+        const { model, result } = event;
+
+        // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
+        if (model.kind === 'collectionType' && model.options.draftAndPublish) {
+          const { id } = result;
+
+          strapi.db.query(RELEASE_ACTION_MODEL_UID).deleteMany({
+            where: {
+              target_type: model.uid,
+              target_id: id,
+            },
+          });
+        }
+      },
+
+      /**
+       * deleteMany hook doesn't return the deleted entries ids
+       * so we need to fetch them before deleting the entries to save the ids on our state
+       */
+      async beforeDeleteMany(event) {
```

Comment:
so I assume when it's a bulk delete the afterDelete event is not called at all?

Reply:
That's exactly the case.